// ngx-braintree/esm5/lib/configure-dropin.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
var ConfigureDropinService = /** @class */ (function () {
    function ConfigureDropinService() {
    }
    /**
     * @param {?} dropinConfig
     * @return {?}
     */
    ConfigureDropinService.prototype.configureCardHolderName = /**
     * @param {?} dropinConfig
     * @return {?}
     */
    function (dropinConfig) {
        dropinConfig.card = {
            cardholderName: {
                required: true
            }
        };
    };
    /**
     * @param {?} dropinConfig
     * @param {?} chargeAmount
     * @param {?} currency
     * @return {?}
     */
    ConfigureDropinService.prototype.configurePaypalCheckout = /**
     * @param {?} dropinConfig
     * @param {?} chargeAmount
     * @param {?} currency
     * @return {?}
     */
    function (dropinConfig, chargeAmount, currency) {
        dropinConfig.paypal = {
            flow: 'checkout',
            amount: chargeAmount,
            currency: currency
        };
    };
    /**
     * @param {?} dropinConfig
     * @return {?}
     */
    ConfigureDropinService.prototype.configurePaypalVault = /**
     * @param {?} dropinConfig
     * @return {?}
     */
    function (dropinConfig) {
        dropinConfig.paypal = {
            flow: 'vault'
        };
    };
    /**
     * @param {?} dropinConfig
     * @param {?} locale
     * @return {?}
     */
    ConfigureDropinService.prototype.configureLocale = /**
     * @param {?} dropinConfig
     * @param {?} locale
     * @return {?}
     */
    function (dropinConfig, locale) {
        dropinConfig.locale = locale;
    };
    ConfigureDropinService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    ConfigureDropinService.ctorParameters = function () { return []; };
    return ConfigureDropinService;
}());
export { ConfigureDropinService };
function ConfigureDropinService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ConfigureDropinService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ConfigureDropinService.ctorParameters;
}
