// ngx-braintree/esm5/lib/ngx-braintree-pay.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Renderer2, ElementRef } from '@angular/core';
import { NgxBraintreeComponent } from './ngx-braintree.component';
var NgxBraintreePayDirective = /** @class */ (function () {
    function NgxBraintreePayDirective(elementRef, renderer, ngxBtComponent) {
        var _this = this;
        this.elementRef = elementRef;
        this.renderer = renderer;
        this.ngxBtComponent = ngxBtComponent;
        // Disable the pay button initially. This will be enabled after the user fills the dropin information.
        this.renderer.setProperty(this.elementRef.nativeElement, 'disabled', true);
        // Subscribe the payButtonStatus event to enable disable the pay button
        this.ngxBtComponent
            .payButtonStatus
            .subscribe(function (status) {
            _this.renderer.setProperty(_this.elementRef.nativeElement, 'disabled', !status);
        });
        // Handle click event for the pay button
        this.renderer.listen(this.elementRef.nativeElement, 'click', function (event) {
            _this.ngxBtComponent.pay();
        });
    }
    /**
     * @return {?}
     */
    NgxBraintreePayDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        // Set the text on the button to the buttonText property that was sent.
        // this.renderer.setProperty(this.elementRef.nativeElement, 'innerText', this.ngxBtComponent.buttonText);
    };
    /**
     * @return {?}
     */
    NgxBraintreePayDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
    };
    NgxBraintreePayDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[ngxPay]'
                },] },
    ];
    /** @nocollapse */
    NgxBraintreePayDirective.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
        { type: NgxBraintreeComponent, },
    ]; };
    return NgxBraintreePayDirective;
}());
export { NgxBraintreePayDirective };
function NgxBraintreePayDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxBraintreePayDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxBraintreePayDirective.ctorParameters;
    /** @type {?} */
    NgxBraintreePayDirective.prototype.elementRef;
    /** @type {?} */
    NgxBraintreePayDirective.prototype.renderer;
    /** @type {?} */
    NgxBraintreePayDirective.prototype.ngxBtComponent;
}
