// ngx-braintree/esm5/lib/ngx-braintree.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxBraintreeService } from './ngx-braintree.service';
import { ConfigureDropinService } from './configure-dropin.service';
import { NgxBraintreeComponent } from './ngx-braintree.component';
import { NgxBraintreeDirective } from './ngx-braintree.directive';
import { NgxBraintreePayDirective } from './ngx-braintree-pay.directive';
var NgxBraintreeModule = /** @class */ (function () {
    function NgxBraintreeModule() {
    }
    NgxBraintreeModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: [
                        NgxBraintreeComponent,
                        NgxBraintreeDirective,
                        NgxBraintreePayDirective
                    ],
                    exports: [
                        NgxBraintreeComponent,
                        NgxBraintreePayDirective
                    ],
                    providers: [
                        { provide: NgxBraintreeService, useClass: NgxBraintreeService },
                        { provide: ConfigureDropinService, useClass: ConfigureDropinService }
                    ]
                },] },
    ];
    return NgxBraintreeModule;
}());
export { NgxBraintreeModule };
function NgxBraintreeModule_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxBraintreeModule.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxBraintreeModule.ctorParameters;
}
