// ngx-braintree/esm2015/lib/configure-dropin.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Injectable } from '@angular/core';
export class ConfigureDropinService {
    constructor() { }
    /**
     * @param {?} dropinConfig
     * @return {?}
     */
    configureCardHolderName(dropinConfig) {
        dropinConfig.card = {
            cardholderName: {
                required: true
            }
        };
    }
    /**
     * @param {?} dropinConfig
     * @param {?} chargeAmount
     * @param {?} currency
     * @return {?}
     */
    configurePaypalCheckout(dropinConfig, chargeAmount, currency) {
        dropinConfig.paypal = {
            flow: 'checkout',
            amount: chargeAmount,
            currency: currency
        };
    }
    /**
     * @param {?} dropinConfig
     * @return {?}
     */
    configurePaypalVault(dropinConfig) {
        dropinConfig.paypal = {
            flow: 'vault'
        };
    }
    /**
     * @param {?} dropinConfig
     * @param {?} locale
     * @return {?}
     */
    configureLocale(dropinConfig, locale) {
        dropinConfig.locale = locale;
    }
}
ConfigureDropinService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
ConfigureDropinService.ctorParameters = () => [];
function ConfigureDropinService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    ConfigureDropinService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    ConfigureDropinService.ctorParameters;
}
