// ngx-braintree/esm2015/lib/ngx-braintree.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Renderer2, Inject } from '@angular/core';
import { DOCUMENT } from '@angular/platform-browser';
export class NgxBraintreeDirective {
    /**
     * @param {?} renderer
     * @param {?} document
     */
    constructor(renderer, document) {
        this.renderer = renderer;
        this.document = document;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.script = this.renderer.createElement('script');
        this.script.type = 'text/javascript';
        this.script.src = 'https://js.braintreegateway.com/web/dropin/1.8.0/js/dropin.min.js';
        this.renderer.appendChild(this.document.body, this.script);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.renderer.removeChild(this.document.body, this.script);
    }
}
NgxBraintreeDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngxBraintreeDirective]'
            },] },
];
/** @nocollapse */
NgxBraintreeDirective.ctorParameters = () => [
    { type: Renderer2, },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] },] },
];
function NgxBraintreeDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxBraintreeDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxBraintreeDirective.ctorParameters;
    /** @type {?} */
    NgxBraintreeDirective.prototype.script;
    /** @type {?} */
    NgxBraintreeDirective.prototype.renderer;
    /** @type {?} */
    NgxBraintreeDirective.prototype.document;
}
