// ngx-braintree/esm2015/lib/ngx-braintree-pay.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Directive, Renderer2, ElementRef } from '@angular/core';
import { NgxBraintreeComponent } from './ngx-braintree.component';
export class NgxBraintreePayDirective {
    /**
     * @param {?} elementRef
     * @param {?} renderer
     * @param {?} ngxBtComponent
     */
    constructor(elementRef, renderer, ngxBtComponent) {
        this.elementRef = elementRef;
        this.renderer = renderer;
        this.ngxBtComponent = ngxBtComponent;
        // Disable the pay button initially. This will be enabled after the user fills the dropin information.
        this.renderer.setProperty(this.elementRef.nativeElement, 'disabled', true);
        // Subscribe the payButtonStatus event to enable disable the pay button
        this.ngxBtComponent
            .payButtonStatus
            .subscribe((status) => {
            this.renderer.setProperty(this.elementRef.nativeElement, 'disabled', !status);
        });
        // Handle click event for the pay button
        this.renderer.listen(this.elementRef.nativeElement, 'click', (event) => {
            this.ngxBtComponent.pay();
        });
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        // Set the text on the button to the buttonText property that was sent.
        // this.renderer.setProperty(this.elementRef.nativeElement, 'innerText', this.ngxBtComponent.buttonText);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
    }
}
NgxBraintreePayDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngxPay]'
            },] },
];
/** @nocollapse */
NgxBraintreePayDirective.ctorParameters = () => [
    { type: ElementRef, },
    { type: Renderer2, },
    { type: NgxBraintreeComponent, },
];
function NgxBraintreePayDirective_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    NgxBraintreePayDirective.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    NgxBraintreePayDirective.ctorParameters;
    /** @type {?} */
    NgxBraintreePayDirective.prototype.elementRef;
    /** @type {?} */
    NgxBraintreePayDirective.prototype.renderer;
    /** @type {?} */
    NgxBraintreePayDirective.prototype.ngxBtComponent;
}
